src/k2/cmd/demo/document_demo.cpp

```diff
@@ -0,0 +1,161 @@
+
+        return _stopPromise.get_future();
+    }
+
+    seastar::future<> start() {
+        _stopped = false;
+        k2::RPC().registerLowTransportMemoryObserver([](const k2::String& ttype, size_t requiredReleaseBytes) {
+            K2WARN("We're low on memory in transport: " << ttype <<", requires release of "<< requiredReleaseBytes << " bytes");
+        });
+
+        return _client.start().then([this] () { return _demo(); })
+        .handle_exception([this](auto exc) {
+            K2ERROR_EXC("Unable to execute benchmark", exc);
+            _stopped = true;
+            return seastar::make_ready_future<>();
+        }).finally([this]() {
+            K2INFO("Done with benchmark");
+            _stopped = true;
+            _stopPromise.set_value();
+            return seastar::make_ready_future<>();
+        });
+    }
+
+private:
+    seastar::future<> _demo() {
+        return _client.makeCollection("demo").discard_result()
+        .then([this] () {
+            dto::Schema schema;
+            schema.name = "demo_schema";
+            schema.version = 1; // We expect to create version 1 of this schema
+            schema.fields = std::vector<dto::DocumentFieldType> {
+                    dto::DocumentFieldType::STRING,
+                    dto::DocumentFieldType::STRING,
+                    dto::DocumentFieldType::UINT32T};
+            schema.fieldNames = std::vector<String>{"LastName", "FirstName", "Balance"};
+            // LastName is our partition key
+            schema.partitionKeyFields = std::vector<uint32_t>{0};
+            // FirstName is our range key
+            schema.rangeKeyFields = std::vector<uint32_t>{1};
+
+            dto::CreateSchemaRequest request{ "demo_schema", std::move(schema) };
+            return RPC().callRPC<dto::CreateSchemaRequest, dto::CreateSchemaResponse>(dto::Verbs::CPO_SCHEMA_CREATE, request, *(RPC().getTXEndpoint(_cpo())), 1s);
+        })
+        .then([this] (auto&& response) {
+            auto& [status, k2response] = response;
+            K2ASSERT(status.is2xxOK(), "Create schema failed");
+
+            return _client.beginTxn(K2TxnOptions{});
+        })
+        .then([this] (K2TxnHandle&& txn) {
+            SerializableDocument writeDoc = _client.makeSerializableDocument("demo", "demo_schema");
+            writeDoc.serializeField<String, dto::DocumentFieldType::STRING>("Baggins", "LastName");
+            writeDoc.serializeField<String, dto::DocumentFieldType::STRING>("Bilbo", "FirstName");
+            writeDoc.serializeField<uint32_t, dto::DocumentFieldType::UINT32T>(777, "Balance");
+
+            return txn.write(std::move(writeDoc))
+            .then([&txn] (WriteResult&& result) {
+                K2ASSERT(result.status.is2xxOK(), "Write failed");
+                return std::move(txn);
+            });
+        })
+        .then([this] (K2TxnHandle&& txn) {
+            SerializableDocument readDoc = _client.makeSerializableDocument("demo", "demo_schema");
```

Comment:
What schema version will this use? On one hand, we want to discover records just based on schema name, not version, to allow for schema updates. On the other, we probably can't handle changes to primary/range key schema since that would mean each version needs different fields for keys...